### PR TITLE
Add a generic Poisson pdf and logpdf

### PR DESCRIPTION
**Pull request by dmbates**
Code much cleaner thanks to a suggestion by @andreasnoack

### CODE REVIEW

File: src/distrs/pois.jl [hunk 1]
```diff
@@ -11,3 +11,12 @@ import .RFunctions:
     poisinvccdf,
     poisinvlogcdf,
     poisinvlogccdf
+
+# generic versions
+poispdf(λ::Real, x::Real) = exp(poislogpdf(λ, x))
+
+function poislogpdf(λ::T, x::T) where {T <: Real}
+    iszero(λ) ? (iszero(x) ? x : T(-Inf)) : x * log(λ) - λ - lgamma(x + 1)
```
Comment (andreasnoack) [inline src/distrs/pois.jl]: StatsFuns has the `xlogy` function exactly for this purpose

Comment (dmbates) [reply src/distrs/pois.jl]: Right.  In fact, I think I added it long, long ago.

**Comment by dmbates** [comment]
CI failures on nightly are b/c the lbeta method for ForwardDiff.Dual uses a digamma which is not accessible.

Can I go ahead and merge?

File: src/distrs/pois.jl [hunk 2]
```diff
@@ -11,3 +11,12 @@ import .RFunctions:
     poisinvccdf,
     poisinvlogcdf,
     poisinvlogccdf
+
+# generic versions
+poispdf(λ::Real, x::Real) = exp(poislogpdf(λ, x))
+
+function poislogpdf(λ::T, x::T) where {T <: Real}
+    iszero(λ) ? (iszero(x) ? x : T(-Inf)) : xlogy(x, λ) - λ - lgamma(x + 1)
```
Comment (andreasnoack) [inline src/distrs/pois.jl]: I don't think the ternaries are needed here when you are using `xlogy`.
```julia
poislogpdf(λ::T, x::T) where {T <: Real} = xlogy(x, λ) - λ - lgamma(x + 1)
```
should be sufficient.

Comment (dmbates) [reply src/distrs/pois.jl]: Right.  Thanks.

**Pull request merged**